[MIG] 11.0 website multi theme

yelizariev opened the pull request:
I've fixed installation problems and didn't make further testing

cc @Yajo

yajo commented:
Could you please fix travis before continuing further reviews? Thanks!

yelizariev commented:
Travis is green now

yajo commented on website_multi_theme/README.rst:
```diff
@@ -0,0 +1,155 @@
+
+  - https://github.com/odoo/odoo/commit/15bf41270d3abb607e7b623b59355594cad170cf
+  - https://github.com/odoo/odoo/commit/7c6714d7fee4125f037ef194f9cff5235a6c5320
+  - https://github.com/odoo/odoo/commit/48fe0a595308722a26afd5361432f24c610b4ba0
```
You can remove this now that v11 includes all of that.

yelizariev commented:
I will do manual tests by the end of this week

lasley commented:
Runbot should be fixed now FYI. I'll be reviewing this one soon as well, thank you for the contribution @yelizariev!

yelizariev commented:
Not sure was @Yajo's comment about runbot itself or about the module, but with latest module updates it should be much easier to try it on runbot -- I've updated demo data and added an instruction to the README file -- just one step is required to finish demo configuration.

yelizariev commented:
Works with Bootswatch Theme

![2017-11-11 12 51 57](https://user-images.githubusercontent.com/186131/32687483-3dd5eb76-c6df-11e7-8bcf-4133ddb97973.png)
![2017-11-11 12 51 48](https://user-images.githubusercontent.com/186131/32687484-3df94080-c6df-11e7-9408-3cc6a96e3067.png)

yajo commented on website_multi_theme/demo/pages.xml:
```diff
@@ -23,11 +39,19 @@
+        <field name="page_id" ref="demo_page" />
+    </record>
+
+    <record id="menu_demo_page2" model="website.menu">
```
Aren't you able to create these 2 menu records here instead of creating them at the top and re-editing them here at the bottom?

yelizariev replied:
I can't. Odoo source has the same approach. The reason in few computed fields that depends on each other. I will send details soon

yelizariev replied:
Well, I don't have the answer at the moment. When I did I got some problem and switched to a "splitting solution" and it began work. But now I tried to do it without splitting and it works too.

The splitting in odoo source may somehow related to splitting website.page and ir.ui.view creation for pages:

https://github.com/odoo/odoo/blob/4ecbacaf59576a22ff45615a5aa5c67244e4fb93/addons/website/data/website_data.xml#L34-L42

https://github.com/odoo/odoo/commit/8baabfb6be9e1db1af0828d705d07a1ec6f0029a#diff-d19576011e13311d9a4dd46879ebd724

yajo commented on website_multi_theme/tests/test_assets.py:
```diff
@@ -40,9 +19,9 @@ def test_localhost(self):
-        response = self.url_open("http://127.0.0.1:%d" % PORT, timeout=30)
+    def test_0_0_0_0(self):
+        """Check 0.0.0.0 downloads its default assets."""
+        response = self.url_open("http://0.0.0.0:%d" % PORT, timeout=30)
```
Wow, TIL that you can use `0.0.0.0` to call to localhost!

tbaden replied:
@yelizariev what was your reason to change this? currently facing travis error on these tests in v10. both 127.0.0.1 and 0.0.0.0 are failing the tests. localhost is working just fine.

yelizariev replied:
It allows using exising ``website`` records. Before the update new website was created
```

            ip = env["website"].create({
                "name": "127.0.0.1",
                "domain": "127.0.0.1",
                "multi_theme_id": False,
            })
```

yelizariev commented:
Shall I squash some of my commits? Is there any guidelines for it?

yajo commented:
Take a look at https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#git

Just squash your migration-related commits in one, while keeping others' history.

yelizariev commented:
Commits are squashed. Original history is here: https://github.com/yelizariev/website/commits/11.0-website_multi_theme-before-squash
